### PR TITLE
Fix/thumbnail improvements

### PR DESCRIPTION
**Pull request by JoshLipps**
### Description
Table Thumbnail improvements

-  Fix the Table Thumbnail to not block on a bunch of image or movie url validations - this is not ideal in a list view
- Add retry behavior with exponential back off to retry  image for thumbnail
- Add Medial Modal loading retries as well and add media load fail state

image

### CODE REVIEW

**Review by isacoder — COMMENTED** [review]
This works correctly, my only concern is that previous implementation will verify that the link is working to preven clicking when url is broken, now you can open the thumbnail even when link is not correct for the example of - Magic Edge Cloud

You can compare with UI Kit page - https://future-standard.github.io/scorer-ui-kit/#/table

Screen Shot 2022-04-27 at 20 18 17

**Comment by JoshLipps** [comment]
> This works correctly, my only concern is that previous implementation will verify that the link is working to preven clicking when url is broken, now you can open the thumbnail even when link is not correct for the example of - Magic Edge Cloud
> 
> You can compare with UI Kit page - https://future-standard.github.io/scorer-ui-kit/#/table
> 
> Screen Shot 2022-04-27 at 20 18 17

This change is intentional. the previous implementation was pretty frustrating to use. Click with no feedback. I think it would be better to open with a proper fail state, compared to never ending loading.

File: src/Tables/atoms/TableRowThumbnail.tsx [hunk 1]
```diff
@@ -84,65 +86,51 @@ interface IProps {
...
+    setImgSrc(image);
+  },[image]);
 
-  useEffect(() => {
-    if ( mediaUrl && mediaType) {
-      verifyPreview(mediaUrl, mediaType);
+  useEffect(()=>{
+    if(imgRef.current && imgRef.current.complete){
+      setShowImage(true);
     }
```
Comment (JoshLipps) [inline src/Tables/atoms/TableRowThumbnail.tsx:118]: This allows for a bug. When src ="" or is cached error this allows for broken images. I will attempt a better solution.

**Review by isacoder — APPROVED** [review]
Worked as expected, thanks for the improvement 🙏

**Pull request merged**